### PR TITLE
fix-cors-policy

### LINKED ISSUE #30: It is not possible to connect from the frontend due to the CORS policy
The CORS policy of the daemon does not allow the GUI to make request to the API. This problem may be a bit complicated to solve, since the port of the wallet GUI changes in a random way and it would probably be good to keep some control mechanisms to prevent any random web page opened with a web browser from making requests to the daemon.

### PR DESCRIPTION
**Pull request by therealssj**
Fixes #30 

Changes:
- Allow requests from any localhost address
- Allow requests for `https://staging.wallet.skycoin.net` and `https://wallet.skycoin.net`

Does this change need to mentioned in CHANGELOG.md?
no

Requires changes in protobuff specs?
no

Requires testing
yes

tests added

### CODE REVIEW

File: src/api/middleware.go [hunk 1]
```diff
@@ -46,10 +47,27 @@ func hostCheck(host string, hostWhitelist []string, handler http.Handler) http.H
 	hostWhitelistMap[fmt.Sprintf("127.0.0.1:%d", port)] = struct{}{}
 	hostWhitelistMap[fmt.Sprintf("localhost:%d", port)] = struct{}{}
 
+	isWhiteListed := func(host string) bool {
+		if _, ok := hostWhitelistMap[host]; ok {
+			return true
+		}
+
+		// allow any localhost origin
+		lregex, err := regexp.Compile(`^https?://localhost|127.0.0.1:\d+$`)
```
Comment (gz-c) [inline src/api/middleware.go]: Wrong regex, the . needs to be escaped:
https://play.golang.org/p/n7ze8VzqI88

File: src/api/http.go [hunk 2]
```diff
@@ -204,12 +204,13 @@ func newServerMux(c muxConfig, usbGateway, emulatorGateway Gatewayer) *http.Serv
...
 	}
+
 	for _, s := range c.hostWhitelist {
 		allowedOrigins = append(allowedOrigins, fmt.Sprintf("http://%s", s))
 	}
 
-	// allow any localhost orogin
-	lregex, err := regexp.Compile(`^https?://localhost:\d+$`)
+	// allow any localhost origin
+	lregex, err := regexp.Compile(`^https?://localhost|127.0.0.1:\d+$`)
```
Comment (gz-c) [inline src/api/http.go]: wrong regex

File: src/api/middleware.go [hunk 3]
```diff
@@ -78,10 +96,31 @@ func originRefererCheck(host string, hostWhitelist []string, handler http.Handle
...
 		hostWhitelistMap[host] = struct{}{}
 	}
 
+	isWhiteListed := func(host string) bool {
+		if _, ok := hostWhitelistMap[host]; ok {
+			return true
+		}
+
+		// allow any localhost origin
+		lregex, err := regexp.Compile(`^https?://localhost|127.0.0.1:\d+$`)
```
Comment (gz-c) [inline src/api/middleware.go]: wrong regex

The regex can be loaded on init, globally

**Comment by Senyoret1** [comment]
@therealssj I tested and still had problems. When I tried using Electron, the request was rejected and the cmd window showed this:
cors
Apparently, there are still some headers that cause problems. The details of the request made by Electrón are:
cors2

**Pull request merged**